### PR TITLE
8 post comment

### PR DESCRIPTION
**Pull request by Daniel17971**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,6 +18,7 @@ function App() {
         <Route path="/user" element={<User />} />
         <Route path="/reviews" element={<AllReview />} />
         <Route path="/reviews/:review_id" element={<Review />} />
+        <Route path="/reviews/:review_id/comments" element={<PostComment />} />
```
Comment (callumu21) [inline src/App.js]: I wouldn't have this as a new route. Adding a comment should be a part of the individual review screen, both for user experience and to save you having redundant endpoints

File: src/components/PostComment.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    if (Object.keys(newComment).length) {
+      postComment(review_id.review_id, newComment)
+        .then((data) => {
+          setErr(null);
+          setNewComment({});
+          setReturnComment(data);
+          setHasPosted(true);
+        })
+        .catch((err) => {
+          setErr("That User doesn't exsist!!!");
```
Comment (callumu21) [inline src/components/PostComment.jsx]: Just be careful of the spelling mistake here - 'That user doesn't exist!!!'. There are also other issues that could happen here, such as an unsuccessful request due to network issues, so consider handling this here as well.

File: src/components/PostComment.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    setNewComment((currentComment) => {
+      return { ...currentComment, username: userName, body };
+    });
+  };
+
+  return (
+    <section className="post-comment">
+      <h2>Comment Here</h2>
+      <form onSubmit={handleSubmit} className="comment-form">
+        <label htmlFor="username-input">Username: </label>
```
Comment (callumu21) [inline src/components/PostComment.jsx]: Users shouldn't have to put in their username - this should instead be handled by having a logged in user across the site. This is also relevant for this as a portfolio piece, as a prospective employer isn't going to be know what valid usernames they can enter to see if this is working

File: src/components/PostComment.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          setReturnComment(data);
+          setHasPosted(true);
+        })
+        .catch((err) => {
+          setErr("That User doesn't exsist!!!");
+        });
+    }
+  }, [newComment, review_id]);
+
+  const handleSubmit = (event) => {
```
Comment (callumu21) [inline src/components/PostComment.jsx:35]: You will need some validation here that a user has actually inputted a comment. Currently, an empty comment can be posted.

**Review by callumu21 — CHANGES_REQUESTED** [review]
A good start, Daniel, but some changes will need to be made before this can be approved:

1. Posting a comment shouldn't need a new route. If you're dead-set on this functionality, that's cool, but it would be much better to have the add form component appear on the individual review page alongside the other comments and review. This is partly for better user experience (if I'm writing a comment, I likely want to see the review and other comments), but also prevents the creation of redundant routes.
2. You should also add a logged-in user, rather than asking someone to supply a username. This can either be hardcoded in state for demo purposes, or you can have a log in screen or button to handle this instead.
3. You need some validation on the text input field for a comment as currently empty comments can be posted.
4. You should also add a reset to the form fields on a successful submission, just in case they want to add more than one comment.
5. You should also consider a way of preventing multiple post attempts being made. Currently, a user can press submit multiple times before the first one is complete. State would be a good way of handling this.

Finally, I would consider looking at some of your styling, just to make sure that presentation is okay on a smaller screen. Currently the form component becomes quite compressed on smaller screens.

Once you've implemented the above, please resubmit this PR and we'll take another look 👍

File: src/components/PostComment.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        .then((data) => {
+          setHasPosted(true);
+          setErr(null);
+          setCommentsList((currentList) => {
+            return [...currentList, data];
+          });
+        })
+        .catch((err) => {
+          console.log("here");
+          setErr("Your not loged in!");
```
Comment (HeatherEver) [inline src/components/PostComment.jsx]: little spelling mistake here in 'logged'

File: src/components/PostComment.jsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      postComment(review_id.review_id, newComment)
+        .then((data) => {
+          setHasPosted(true);
+          setErr(null);
+          setCommentsList((currentList) => {
+            return [...currentList, data];
+          });
+        })
+        .catch((err) => {
+          console.log("here");
```
Comment (HeatherEver) [inline src/components/PostComment.jsx]: remove console logs

**Review by HeatherEver — COMMENTED** [review]
This is good work Daniel but there's still a couple of things to look at here... 
- when i log in as a user there's no kind of confirmation to say it's been successful, which as a user would be nice
- when i post a comment both messages of 'you are not logged in' and 'post successful' appear on the screen (see below)
- the new comment also goes to the back of the comments which is good when there's only one page but when there's multiple pages my comment goes to the very back which isn't very user friendly 
- although the CSS isn't too important right now, it's worth noting that when the screen is changed size some things start to get a bit weird (see below) 

Screenshot 2023-04-19 at 11 38 50


Screenshot 2023-04-19 at 11 28 58


Screenshot 2023-04-19 at 11 28 47

**Pull request merged**